src/app/client/src/app/modules/core/components/main-header/main-header.component.ts

```diff
@@ -100,9 +100,26 @@ export class MainHeaderComponent implements OnInit, OnDestroy {
     this.permissionService = permissionService;
     this.userService = userService;
     this.tenantService = tenantService;
-  }
+   }
 
   ngOnInit() {
+    this.router.events.pipe(filter(event => event instanceof NavigationEnd))
+      .subscribe(event => {
+        let currentRoute = this.activatedRoute.root;
+        while (currentRoute.children.length > 0) {
```

Comment:
what happens if children is null?

Reply:
currentRoute.children is an activatedRoute, so it will not be null.